tests/extensions/test_table.py

```diff
@@ -59,3 +59,21 @@ def test_should_raise_exception_when_passing_invalid_extension_object(
             TableExtension.ext,
             object(),
         )
+
+    def test_item_with_table_extension_is_serilalizable_and_roundtrips(self) -> None:
+        item = pystac.Item.from_file(self.example_uri)
+        # add column metadata
+        tab_ext = TableExtension.ext(item, add_if_missing=True)
+        columns = [
+            Column({"name": "col_1", "type": "str"}),
+            Column({"name": "col_2", "type": "byte_array"}),
+        ]
+        tab_ext.columns = columns
+        item.save_object()
```

Comment:
One other change please -- currently this overwrites the test data file, leading to a git diff after running tests. Coupled with the change to the free function (https://github.com/stac-utils/pystac/pull/992/files#r1107712387), could you use [tmp_path](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html) to save the new item to a temporary location, so we don't modify the code tree? Thanks!